lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -400,7 +400,7 @@ def compile_method(symbol_name, constant, method)
             end
           end.join(', ')
 
-          method_name.prepend(constant.singleton_class? ? 'self.' : '')
+          method_name = "#{constant.singleton_class? ? 'self.' : ''}#{method_name}"
```

Comment:
```suggestion
          method_name = "#{'self.' if constant.singleton_class?}#{method_name}"
```
would be even nicer, since:

```ruby
pry(main)> "#{'hello' if false}"
=> ""
```